sdk/core/azure-core/tests/test_authentication.py

```diff
@@ -79,3 +80,24 @@ def test_bearer_policy_enforces_tls():
     pipeline = Pipeline(transport=Mock(), policies=[BearerTokenCredentialPolicy(credential, "scope")])
     with pytest.raises(ServiceRequestError):
         pipeline.run(HttpRequest("GET", "http://not.secure"))
+
+
+@pytest.mark.skipif(azure.core.__version__ >= "2", reason="this test applies only to azure-core 1.x")
```

Comment:
Don't skip it, indeed this test is about testing that any version after 1.2.x still works with code written based on azure-core 1.0.0. And because there is no limit in our backward compat, this test MUST work with all future release version of azure-core, even 1.42.0

Reply:
@lmazuel isn't the skip condition for excluding on *2.x* or higher? (presumably a major number bump explicitly indicated compatibility breaking changes)

Reply:
@bryevdv ho you're right, I got it wrong since the test was about troubles in 1.2 and got the wrong too.
Still, I'm not sure this skip worth it, if we (ever) release a 2.x (which I doubt), test will likely have to be refactored anyway.
But I'm not against, I'm just not "for" either :). Removing the blocking at least

Reply:
It is not a big deal. If we move to 2.0, we need to update lots of tests. :)

Reply:
My vision is of our grandchildren beginning their investigation asking "do we still need this skipped test" rather than "why is this weird test failing" 😉

I agree it's a minor point.